transactionverifier/transactionVerifier.go

```diff
@@ -0,0 +1,211 @@
+package transactionverifier
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"errors"
+	"github.com/algorand/go-algorand-sdk/client/v2/common/models"
+	"github.com/algorand/go-algorand-sdk/stateproofs/transactionverificationtypes"
+	"github.com/algorand/go-algorand-sdk/types"
+)
+
+var (
+	ErrUnsupportedHashFunction      = errors.New("proof hash function is unsupported")
+	ErrProofLengthTreeDepthMismatch = errors.New("proof length and tree depth do not match")
+	ErrRootMismatch                 = errors.New("root mismatch")
+	ErrInvalidTreeDepth             = errors.New("invalid tree depth")
+	ErrIndexDepthMismatch           = errors.New("node index is not smaller than 2^depth")
+	ErrInvalidPosition              = errors.New("invalid position for node")
+)
+
+type NodePosition int
+
+const (
+	leftChild NodePosition = iota
+	rightChild
+)
+
+// computeTransactionLeaf receives the transaction ID and the signed transaction in block's hash, and computes
+// the leaf of the vector commitment associated with the transaction.
+// Parameters:
+// txId - the Sha256 hash of the msgpacked transaction.
+// stibHash - the Sha256 of the msgpacked transaction as it's saved in the block.
+func computeTransactionLeaf(txId types.Digest, stibHash types.Digest) types.Digest {
+	leafDomainSeparator := []byte(transactionverificationtypes.TxnMerkleLeaf)
+	leafData := leafDomainSeparator
```

Comment:
why have 2 new vars for this if we just append?

Reply:
Readability - it's very easy to understand how leafData is built when using multiple variables. I've tweaked the code a bit even more, hopefully it's even clearer now.